### PR TITLE
RavenDB-7801: Fix usages of ConcurrentDictionary that could cause exc…

### PR DESCRIPTION
**Pull request by jbayardo**
…essive locking and add warnings

### CODE REVIEW

**Comment by jbayardo** [comment]
Found a regression while running SlowTests, please do not merge yet.

EDIT: Turns out I can't know if my code works or not because of [RavenDB-8061](http://issues.hibernatingrhinos.com/issue/RavenDB-8061) causing slow indexes to fail the tests. If CI passes the tests please feel free to merge, I will work on fixing 8061

File: src/Raven.Client/Util/AtomicDictionary.cs [hunk 1]
```diff
@@ -29,6 +29,9 @@ public AtomicDictionary(IEqualityComparer<string> comparer)
             _locks = new ConcurrentDictionary<string, object>(comparer);
         }
 
+        /// <summary>
+        /// This locks the entire dictionary. Use carefully.
+        /// </summary>
         public IEnumerable<TVal> Values => _items.Values;
```
Comment (ayende) [inline src/Raven.Client/Util/AtomicDictionary.cs:35]: Are there any usages of this, if not, just kill the prop.
Also, I think we can just use `ConcurrentDictionary<string,Lazy<T>>` here instead, no?

File: src/Raven.Server/Documents/Operations/Operations.cs [hunk 2]
```diff
@@ -236,15 +236,9 @@ public void Dispose(ExceptionAggregator exceptionAggregator)
...
-        {
-            return _active.Values.Union(_completed.Values);
-        }
+        public IEnumerable<Operation> GetAll() => _active.Values.Union(_completed.Values);
 
-        public ICollection<Operation> GetActive()
-        {
-            return _active.Values;
-        }
+        public ICollection<Operation> GetActive() => _active.Values;
```
Comment (ayende) [inline src/Raven.Server/Documents/Operations/Operations.cs:241]: Shouldn't we iterate and return the results instead?

File: src/Raven.Server/Documents/Operations/Operations.cs [hunk 3]
```diff
@@ -236,15 +236,9 @@ public void Dispose(ExceptionAggregator exceptionAggregator)
             _completed.Clear();
         }
 
-        public IEnumerable<Operation> GetAll()
-        {
-            return _active.Values.Union(_completed.Values);
-        }
+        public IEnumerable<Operation> GetAll() => _active.Values.Union(_completed.Values);
```
Comment (ayende) [inline src/Raven.Server/Documents/Operations/Operations.cs:239]: Shouldn't we iterate and return the results?

**Pull request merged**